wpilibc/src/main/native/include/frc/simulation/DifferentialDrivetrainSim.h

```diff
@@ -0,0 +1,218 @@
+   * output over input. This must be the same ratio as the ratio used to
+   * identify or create the drivetrainPlant.
+   * @param J           The moment of inertia of the drivetrain about its
+   * center.
+   * @param mass        The mass of the drivebase.
+   * @param wheelRadius The radius of the wheels on the drivetrain.
+   * @param trackWidth  The robot's track width, or distance between left and
+   * right wheels.
+   */
+  DifferentialDrivetrainSim(frc::DCMotor driveMotor, double gearing,
+                            units::kilogram_square_meter_t J,
+                            units::kilogram_t mass, units::meter_t wheelRadius,
+                            units::meter_t trackWidth);
+
+  /**
+   * Set the applied voltage to the drivetrain. Note that positive voltage must
+   * make that side of the drivetrain travel forward (+X).
+   *
+   * @param leftVoltage  The left voltage.
+   * @param rightVoltage The right voltage.
+   */
+  void SetInputs(units::volt_t leftVoltage, units::volt_t rightVoltage);
+
+  /**
+   * Set the gearing reduction on the drivetrain. This is commonly used for
+   * shifting drivetrains.
+   *
+   * @param newGearing The new gear ratio, as output over input.
+   */
+  void SetGearing(double newGearing);
+
+  /**
+   * Updates the simulation.
+   *
+   * @param dt The time that's passed since the last {@link #update(double)}
+   * call.
+   */
+  void Update(units::second_t dt);
+
+  /**
+   * Returns an element of the state vector.
+   *
+   * @param state The row of the state vector.
+   */
+  double GetState(int state) const;
+
+  /**
+   * Get the current gearing reduction of the drivetrain, as output over input.
+   */
+  double GetGearing() const;
+
+  /**
+   * Returns the current state vector x.
+   */
+  Eigen::Matrix<double, 7, 1> GetState() const;
+
+  /**
+   * Get the estimated direction the robot is pointing. Note that this angle is
+   * counterclockwise-positive, while most gyros are clockwise positive.
+   */
+  Rotation2d GetHeading() const;
+
+  /**
+   * Returns the current estimated position.
+   */
+  Pose2d GetEstimatedPosition() const;
+
+  /**
+   * Returns the currently drawn current.
+   */
+  units::ampere_t GetCurrentDraw() const;
+
+  Eigen::Matrix<double, 7, 1> Dynamics(const Eigen::Matrix<double, 7, 1>& x,
+                                       const Eigen::Matrix<double, 2, 1>& u);
+
+  class State {
+   public:
+    static constexpr int kX = 0;
+    static constexpr int kY = 1;
+    static constexpr int kHeading = 2;
+    static constexpr int kLeftVelocity = 3;
+    static constexpr int kRightVelocity = 4;
+    static constexpr int kLeftPosition = 5;
+    static constexpr int kRightPosition = 6;
+  };
+
+  /**
+   * Represents a gearing option of the Toughbox mini.
+   * 12.75:1 -- 14:50 and 14:50
+   * 10.71 -- 14:50 and 16:48
+   * 8.45:1 -- 14:50 and 19:45
+   * 7.31 -- 14:50 and 21:43
+   * 5.95: 14:50 and 24:40
```

Comment:
14/50 * 14/50 does not equal 12.75. The notation here isn't consistent in general.
```suggestion
   * 12.75:1 -- 14:50 and 14:50
   * 10.71 -- 14:50 and 16:48
   * 8.45:1 -- 14:50 and 19:45
   * 7.31 -- 14:50 and 21:43
   * 5.95 -- 14:50 and 24:40
```